configuration/configurator/ConfiguratorMain.cpp

```diff
@@ -84,7 +84,7 @@ void usage()
     memset(pBasePath, 0, sizeof(pBasePath));
     memset(pEnvXMLPath, 0, sizeof(pEnvXMLPath));
 
-    strncpy(pTargetDocExt, pDefaultDocExt, sizeof(pTargetDocExt));
+    strcpy(pTargetDocExt, pDefaultDocExt);
```

Comment:
This is simply swapping one unsafe behaviour (the resulting string may be unterminated) with another (the resulting string will be terminated but adjacent data will be overwritten).

There is no reason to use fixed size buffers here. Change pTargetDocExt into a StringBuffer